### PR TITLE
Remove effective gas price from EIP-1559 receipt

### PR DESCRIPTION
**Pull request by lightclient**
I don't believe this extra field is useful in a consensus structure. Each field in the receipt can currently *only* be determined by reexecuting the transaction. `effectiveGasPrice` does not require reexecution, only the transaction's `feeCap` and `tip` in addition to the block's `baseFee`. This generally requires 1 additional RPC call to get the block `baseFee`, on top of what is done today (just `eth.getTransaction` today). Client could optimize this further by indexing transaction in relation to the block `baseFee`.

### CODE REVIEW

**Comment by eip-automerger** [comment]
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):

 - EIP 1559 requires approval from one of (@mslipper, @afdudley, @econoar, @i-norden, @abdelhamidbakhta, @vbuterin)

**Comment by AFDudley** [comment]
This makes sense. Is there a way we can make sure clients add the new index?

**Comment by lightclient** [comment]
@AFDudley we could modify the RPC spec to return `effectiveGasPrice` as part of `eth.getTransaction` and let clients decide how to implement. In the beginning, they may read the tx from disk and then the block to get the `baseFee` and construct the response, but in the future they could also add a `txhash => effectiveGasPrice` index if desired. The RPC spec is out of my domain of expertise though, so I would defer to those who most regularly interact with it.

**Review by abdelhamidbakhta — APPROVED** [review]
Lgtm

**Pull request merged**
